### PR TITLE
Fix tests on linux

### PR DESCRIPTION
**Pull request by Utagai**
So this should fix the problem on Linux. On Windows and nicer OSes, we fail early anyways and don't pay the cost of extra recursive calls to the `stat` syscall. On Linux, we make such recursive calls only as much as we need until we get a `true` _OR_ a non-`nil` error.

### CODE REVIEW

File: pkg/sh/exec.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func RunOutput(ctx *task.Context, name string, args ...string) (string, error) {
 // RunCmd runs the provided command.
 func RunCmd(ctx *task.Context, cmd *exec.Cmd) error {
 	LogCmd(ctx, cmd)
-	if ctx.Verbose && cmd.Stdout == nil {
+	if cmd.Stdout == nil {
```
Comment (Utagai) [inline pkg/sh/exec.go]: I am not sure if you actually want this change. Perhaps you would prefer the usage to require `-v` for test output.

Comment (craiggwilson) [reply pkg/sh/exec.go]: yeah, the intent here is that -v is required if you want to see the content.

**Review by craiggwilson — APPROVED** [review]
other than putting the ctx.Verbose back, feel free to push away.

**Comment by Utagai** [comment]
Seems like I can't merge despite now having push access. However, this isn't super urgent so feel free to merge after the weekend.

**Review by craiggwilson — APPROVED** [review]

**Pull request merged**